perlock-core/src/main/java/name/mitterdorfer/perlock/SinglePathChangeListener.java

```diff
@@ -0,0 +1,18 @@
+package name.mitterdorfer.perlock;
+
+import java.nio.file.Path;
+import java.nio.file.WatchEvent;
+
+/**
+ * Callback interface for notifying the user when the chosen file changes.
+ * Suited for use as a functional interface
+ *
+ * <p>Implementation note: A <code>PathChangeListener</code> will be called from the internal <code>PathWatcher</code>
+ * thread. This implies that callback methods should return reasonably fast and offload heavy lifting to a dedicated
+ * thread. Otherwise, events may be lost.</p>
+ */
+public interface SinglePathChangeListener {
```

Comment:
Thinking about it some more has sort of opened a can of worms. Perhaps the API should just be 
```java
void onPathChanged(WatchEvent<Path> event);
```
We could potentially just return the original `WatchEvent`, however this would mean that the `context()` would return a relative `Path`.
One option could be implementing `WatchEvent` and setting the `context()` to a resolved path, but this would be against the `WatchEvent` documentation.
One more option would be to expose our own version of a watch event, like the one you have at `AssertingPathChangeListener.Event` that is a bit cleaner to work with.

One other thing that might be worth considering from an API point of view is `OVERFLOW` events.
Currently perlock silently ignores them, but I think there should be some way for the user to be made aware of these.